lib/eventception/dispatcher.rb

```diff
@@ -98,20 +98,20 @@ def listeners_for?(event_name:)
     # priority::
     #   The higher this value, the earlier an event listener will be triggered in the chain (defaults to 0)
     #
-    def add_listener(event_name:, listener:, priority: 0)
-      event_listeners[event_name][priority] << listener
+    def add_listener(event_name:, listener_handler:, priority: 0)
```

Comment:
I was thinking: what do you think of avoiding exposing the `EventHandler` object outside the library? E.g. we would accept a `listener` and `method` on the interfaces, and would create an `EventHandler` from it?

That way is one less class that needs to be known to clients.